src/System.Diagnostics.Process/src/System/Diagnostics/Process.Windows.cs

```diff
@@ -636,43 +658,71 @@ private bool StartCore(ProcessStartInfo startInfo)
 
                     if (startInfo.UserName.Length != 0)
                     {
+                        if(startInfo.Password != null && startInfo.PasswordInClearText != null)
+                        {
+                            throw new ArgumentException(SR.CantSetDuplicatePassword);
+                        }
+
                         Interop.mincore.LogonFlags logonFlags = (Interop.mincore.LogonFlags)0;
                         if (startInfo.LoadUserProfile)
                         {
                             logonFlags = Interop.mincore.LogonFlags.LOGON_WITH_PROFILE;
                         }
 
-
-                        try { }
-                        finally
+                        string password = string.Empty;
+                        IntPtr secureStringPtr = IntPtr.Zero;
+                        try
                         {
-                            retVal = Interop.mincore.CreateProcessWithLogonW(
-                                    startInfo.UserName,
-                                    startInfo.Domain,
-                                    startInfo.PasswordInClearText,
-                                    logonFlags,
-                                    null,            // we don't need this since all the info is in commandLine
-                                    commandLine,
-                                    creationFlags,
-                                    environmentPtr,
-                                    workingDirectory,
-                                    startupInfo,        // pointer to STARTUPINFO
-                                    processInfo         // pointer to PROCESS_INFORMATION
-                                );
+                            if(startInfo.Password != null)
+                            {
+                                secureStringPtr = Marshal.SecureStringToGlobalAllocUnicode(startInfo.Password);
+                                password = Marshal.PtrToStringUni(secureStringPtr);
```

Comment:
Yes, we should not use a string.  The limited value of SecureString comes from keeping the data out of the managed heap, and converting it to a string defeats that.  Using IntPtr is fine.  For SecureString, SecureStringToGlobalAllocUnicode can be used as you've done, and for a regular string, you can use unsafe code with fixed to get a ```char*``` that can then be cast to IntPtr.